src/connectivity/mod.rs

```diff
@@ -266,6 +270,65 @@ pub fn is_weakly_connected(graph: &digraph::PyDiGraph) -> PyResult<bool> {
     Ok(weakly_connected_components(graph)[0].len() == graph.graph.node_count())
 }
 
+/// Check if the graph is semi connected
+///
+/// :param PyDiGraph graph: The graph to check if it is semi connected
+///
+/// :returns: Whether the graph is semi connected or not
+/// :rtype: bool
+///
+/// :raises NullGraph: If an empty graph is passed in
+#[pyfunction]
+#[pyo3(text_signature = "(graph, /)")]
+pub fn is_semi_connected(graph: &digraph::PyDiGraph) -> PyResult<bool> {
+    if graph.graph.node_count() == 0 {
+        return Err(NullGraph::new_err("Invalid operation on a NullGraph"));
+    }
+
+    let mut temp_graph = DiGraph::new();
+    let mut node_map = Vec::new();
+
+    for _node in graph.graph.node_indices() {
+        node_map.push(temp_graph.add_node(()));
+    }
+
+    for edge in graph.graph.edge_indices() {
+        let (source, target) = graph.graph.edge_endpoints(edge).unwrap();
+        temp_graph.add_edge(node_map[source.index()], node_map[target.index()], ());
+    }
+
+    let condensed = condensation(temp_graph, false);
```

Comment:
```suggestion
    let condensed = condensation(temp_graph, true);
```
I think we shouldn't let self-loops impact and multi-edges impact the topological sort